### PR TITLE
feat: implement insights team feature

### LINKED ISSUE #975: Feature: Make the teams UI feature work
### Type of feature

🍕 Feature

### Current behavior

This is dependent on the completion of https://github.com/open-sauced/insights/issues/974 & https://github.com/open-sauced/ops-backend/issues/3

### Suggested solution

Users should be associated to insight pages, and insight pages will become the meeting and collaboration space for users. 

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by OgDev-01**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
This PR:

- Adds teams invite, update and delete implementation
- refactors the select options to accept objects with `name` and `value` key pairs

>Note:

The select needs a bit of work to be responsive atm... It overflows on mobile. We need to replace the select with Radix select that we already have in the design system. I'll open an issue on this to refactor the current radix implementation

## Related Tickets & Documents
Closes #975

## Mobile & Desktop Screenshots/Recordings

image

image

image

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by brandonroberts** [comment]
@OgDev-01 the `user` has been added to the insight information https://beta.api.opensauced.pizza/#/Insights%20service/findInsightPageById

**Comment by brandonroberts** [comment]
@OgDev-01 there's a build failure on the deploy preview

**Comment by OgDev-01** [comment]
> @OgDev-01 there's a build failure on the deploy preview

Fixed ✅

File: lib/hooks/useInsightMembers.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import useSWR, { Fetcher } from "swr";
+import useSupabaseAuth from "./useSupabaseAuth";
+import publicApiFetcher from "lib/utils/public-api-fetcher";
+interface PaginatedInsightMembers {
+  data: DbInsightMember[];
+  meta: Meta;
+}
+
+const useInsightMembers = (page_id: number) => {
```
Comment (brandonroberts) [inline lib/hooks/useInsightMembers.ts:9]: Let's try to keep the naming/casing for variables in the codebase consistent

```suggestion
const useInsightMembers = (insightId: number) => {
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: components/molecules/TeamMemberRow/team-member-row.tsx [hunk 2]
```diff
@@ -7,39 +7,64 @@ import { TeamMemberData } from "../TeamMembersConfig/team-members-config";
...
+const TeamMemberRow = ({ className, name, avatarUrl, access, email, onDelete, onUpdate, id }: TeamMemberRowProps) => {
   const [isMenuOpen, setIsMenuOpen] = useState(false);
 
-  const pending = role == "pending";
+  const pending = access == "pending";
 
-  const handleRoleChange = (role: string) => {};
+  const handleRoleChange = async (role: string) => {
+    setIsMenuOpen(false);
+    if (role === "remove") {
```
[the remark this answers is not in the document]
Comment (OgDev-01) [reply components/molecules/TeamMemberRow/team-member-row.tsx:30]: I have disabled the select trigger when the access is owner for now..

**Comment by Deadreyo** [comment]
image
For some reason my name is displayed as undefined.

File: components/molecules/FilterCardSelect/filter-card-select.tsx [hunk 3]
```diff
@@ -62,22 +66,46 @@ const FilterCardSelect: React.FC<FilterCardSelectProps> = ({ selected: filterNam
...
+            src={
+              icon === "topic"
+                ? icons.topic.src
+                : icon === "org"
+                  ? icons.org.src
+                  : icon === "contributor"
+                    ? icons.contributor.src
+                    : icon === "repo"
+                      ? icons.repo.src
+                      : icons.topic.src
```
Comment (Deadreyo) [inline components/molecules/FilterCardSelect/filter-card-select.tsx:97]: This needs a refractor. Could be a switch-case in the body of the function and using a variable from there, or using the `icon` prop as a key in `icons` as in `icons[icon] ? icons[icon].src : icons.topic.src`. Etc.

File: components/molecules/TeamMemberRow/team-member-row.tsx [hunk 4]
```diff
@@ -7,39 +7,72 @@ import { TeamMemberData } from "../TeamMembersConfig/team-members-config";
 
 interface TeamMemberRowProps extends TeamMemberData {
   className?: string;
+  onDelete: Function;
+  onUpdate: Function;
```
Comment (Deadreyo) [inline components/molecules/TeamMemberRow/team-member-row.tsx:11]: From usage I understood that `onDelete` is a function that takes 1 argument, and `onUpdate` is a function that takes 2 arguments. This should be clearly defined in the type and have a strict structure required from the calling function. `Function` is too generic.

File: components/molecules/TeamMembersConfig/team-members-config.tsx [hunk 5]
```diff
@@ -1,30 +1,89 @@
...
+import { useState } from "react";
+import { validateEmail } from "lib/utils/validate-email";
+import { useToast } from "lib/hooks/useToast";
 
 interface TeamMembersConfigProps {
   className?: string;
   members: TeamMemberData[];
+  onAddMember: Function;
+  onDeleteMember: Function;
+  onUpdateMember: Function;
```
Comment (Deadreyo) [inline components/molecules/TeamMembersConfig/team-members-config.tsx:13]: Same for these. Type is too generic and doesn't represent the input or output of the function.

**Review by Deadreyo — CHANGES_REQUESTED** [review]

**Comment by OgDev-01** [comment]
> image For some reason my name is displayed as undefined.

@brandonroberts any idea why this would occur?

**Comment by brandonroberts** [comment]
> > image For some reason my name is displayed as undefined.
> 
> @brandonroberts any idea why this would occur?

The user doesn't have a `named` defined in the database

image

We should fallback to the `login` if its not defined

**Comment by OgDev-01** [comment]
@Deadreyo can you test again to confirm the undefined issue has been handled?

**Review by Deadreyo — APPROVED** [review]
Yeah it is fixed now. LGTM! 🚀

**Review by brandonroberts — APPROVED** [review]
LGTM

**Review by bdougie — APPROVED** [review]
Functionally checked it out and works for me.

**Pull request merged**